modules/atomic-widgets/assets/js/editor/utils/regenerate-local-style-ids.js

```diff
@@ -59,5 +59,7 @@ export function regenerateLocalStyleIds( container ) {
 
 	const styledElements = allElements.filter( ( element ) => Object.keys( element.model.get( 'styles' ) ?? {} ).length > 0 );
 
-	styledElements?.forEach( updateStyleId );
+	// Should be executed in reverse, so that the most outer element gets the updates last,
+	// as the parent settings update seems to override its descendants' changes
+	styledElements?.reverse().forEach( updateStyleId );
```

Comment:
Maybe extra to function with meaningful name:

```suggestion
	updateStyleIdInReverseOrderSoParentUpdatesLast();
```

Reply:
done